Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs

```diff
@@ -0,0 +1,230 @@
+            if (!Resolve(used, ref utensil) || (utensil.Types & UtensilType.Knife) == 0)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-knife-needed"), uid, Filter.Entities(user));
+                return false;
+            }
+
+            component.MeatParts--;
+
+            if (!string.IsNullOrEmpty(component.MeatPrototype))
+            {
+                var meat = EntityManager.SpawnEntity(component.MeatPrototype, Transform(uid).Coordinates);
+                EntityManager.GetComponent<MetaDataComponent>(meat).EntityName = component.MeatName;
+            }
+
+            if (component.MeatParts != 0)
+            {
+                _popupSystem.PopupEntity(component.MeatSource1p, uid, Filter.Entities(user));
+            }
+            else
+            {
+                UpdateAppearance(uid, null, component);
+                _popupSystem.PopupEntity(component.MeatSource0, uid, Filter.Entities(user));
+            }
+
+            return true;
+        }
+
+        private void UpdateAppearance(EntityUid uid, AppearanceComponent? appearance = null, KitchenSpikeComponent? component = null)
+        {
+            if (!Resolve(uid, ref component, ref appearance, false))
+                return;
+            
+            appearance.SetData(KitchenSpikeVisuals.Status, (component.MeatParts > 0) ? KitchenSpikeStatus.Bloody : KitchenSpikeStatus.Empty);
+        }
+
+        private bool Spikeable(EntityUid uid, EntityUid userUid, EntityUid victimUid,
+            KitchenSpikeComponent? component = null, SharedButcherableComponent? butcherable = null)
+        {
+            if (!Resolve(uid, ref component))
+                return false;
+
+            if (component.MeatParts > 0)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-deny-collect", ("this", uid)), uid, Filter.Entities(userUid));
+                return false;
+            }
+
+            if (!Resolve(victimUid, ref butcherable) || butcherable.MeatPrototype == null)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-deny-butcher", ("victim", victimUid), ("this", uid)), victimUid, Filter.Entities(userUid));
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool TrySpike(EntityUid uid, EntityUid userUid, EntityUid victimUid, KitchenSpikeComponent? component = null,
+            SharedButcherableComponent? butcherable = null, MobStateComponent? mobState = null)
+        {
+            if (!Resolve(uid, ref component) || component.InUse ||
+                !Resolve(victimUid, ref butcherable) || butcherable.BeingButchered)
+                return false;
+
+            // THE WHAT? (again)
+            // Prevent dead from being spiked TODO: Maybe remove when rounds can be played and DOT is implemented
+            if (Resolve(victimUid, ref mobState) &&
+                !mobState.IsDead())
```

Comment:
As above I think some of the resolves are a bit suss.

Reply:
For the first two: I guess it should log that something invoked "TrySpike" on Entities without appropriate components, for other resolvers with popups - yea no log